frontend/packages/ceph-storage-plugin/src/components/modals/clone-pvc-modal/clone-pvc-modal.tsx

```diff
@@ -0,0 +1,174 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Form, FormGroup, TextInput } from '@patternfly/react-core';
+import {
+  convertToBaseValue,
+  humanizeBinaryBytes,
+  LoadingInline,
+  ResourceIcon,
+  withHandlePromise,
+} from '@console/internal/components/utils/index';
+import {
+  createModalLauncher,
+  ModalBody,
+  ModalSubmitFooter,
+  ModalTitle,
+  ModalComponentProps,
+} from '@console/internal/components/factory';
+import { k8sCreate, K8sResourceKind } from '@console/internal/module/k8s';
+import { PersistentVolumeClaimModel } from '@console/internal/models/index';
+import { HandlePromiseProps } from '@console/internal/components/utils/promise-component';
+import { usePrometheusPoll } from '@console/internal/components/graphs/prometheus-poll-hook';
+import { PrometheusEndpoint } from '@console/internal/components/graphs/helpers';
+import { getInstantVectorStats } from '@console/internal/components/graphs/utils';
+import { DataPoint } from '@console/internal/components/graphs/index';
+import { getPVCUsedCapacityQuery } from '../../../constants/queries';
+import './_clone-pvc-modal.scss';
+
+const accessModeRadios = Object.freeze({
+  ReadWriteOnce: 'Single User (RWO)',
+  ReadWriteMany: 'Shared Access (RWX)',
+  ReadOnlyMany: 'Read Only (ROX)',
+});
+
+const ClonePVCModal = withHandlePromise((props: ClonePVCModalProps) => {
+  const { close, cancel, resource, handlePromise, errorMessage, inProgress } = props;
+  const pvcName: string = resource.metadata.name;
+  const [submitDisabled, setSubmitDisabled] = React.useState(false);
+  const [clonePVCName, setClonePVCName] = React.useState(`${pvcName}-clone`);
+
+  const pvcUsedCapacityQuery: string = getPVCUsedCapacityQuery(pvcName);
+
+  const [response, error, loading] = usePrometheusPoll({
```

Comment:
You need to specify `namespace` to `usePrometheusPoll` or normal users won't be able to run this query.